sycl/source/detail/queue_impl.hpp

```diff
@@ -534,6 +562,14 @@ class queue_impl {
   std::mutex MLastEventMtx;
 
   const bool MIsInorder;
+
+public:
+  // Queue constructed with the discard_events property
+  const bool MDiscardEvents;
+
+private:
+  // The presence of support for DiscardEvents from the queue
+  const bool MHasDiscardEventsSupport;
```

Comment:
Could you please clarify what this support means?

Reply:
`MHasDiscardEventsSupport` serves to answer the question: can we discard the low-level event using this queue? The variable answers this question in terms of queue support, because the final decision about discarding is decided before the operation itself, since it is necessary to check whether this operation uses accessors, and also in the case of a kernel, it is necessary to check the use of assert() statement.

For example, you can create a queue with  `discard_events` property, but without `in_order` property then the queue doesn't support discarding.
Another example, you can create a queue with  `discard_events`  and  `in_order` properties, but run it on level_zero backend(which doesn't support discarding events yet) then the queue doesn't support discarding.

Reply:
Ok, so this flags says if we can discard events based on a queue "setup" which will be common for all kernels submitted to the queue. And as you said this is a must condition for discarding, but even if it's true, in some cases, we won't be able to discard events.

It would be nice if this information is added as a comment.

Reply:
Ok, done.